docs/proposals/proposal-002-run.md

```diff
@@ -0,0 +1,313 @@
+
+#### Use Case 2: A GitHub Action job using a workflow defined in a _different_ repository
+
+We want to accommodate different methods of setting up the tests depending on the CNCF project. Given this, the test workflow containing the test instructions could be defined in a different repository. In this case, the `test_path` would be, for example, `falcosecurity/cncf-green-review-testing/.github/workflows/workflow.yml@v1`.
+
+![Pipeline run](diagrams/pipeline-run.png "An example pipeline run")
+
+### Test instructions
+
+The test workflow which contains the test instructions may look like the following:
+
+```yaml
+# .github/workflows/tests/falco-test-workflow.yaml
+jobs:
+  stress-ng-test:
+    runs-on: ubuntu-latest
+    steps:
+      - run: |
+          # the action to take here depends on the Functional Unit of the CNCF project. wait for amount of time, for resources
+          kubectl apply -f https://raw.githubusercontent.com/falcosecurity/cncf-green-review-testing/main/kustomize/falco-driver/ebpf/stress-ng.yaml
+          wait 15m
+          kubectl delete -f https://raw.githubusercontent.com/falcosecurity/cncf-green-review-testing/main/kustomize/falco-driver/ebpf/stress-ng.yaml
+  # other Falco tests:
+  # - redis-test e.g. https://github.com/falcosecurity/cncf-green-review-testing/blob/main/kustomize/falco-driver/ebpf/redis.yaml
+  # - event-generator-test e.g. https://github.com/falcosecurity/cncf-green-review-testing/blob/main/kustomize/falco-driver/ebpf/falco-event-generator.yaml
+  # TODO: should each test be a workflow or a job in a single workflow? as in, one test workflow per cncf project or multiple workflows per cncf project? TBD
+```
+
+The job has test instructions to apply the upstream Kubernetes manifest which contains a `while` loop that runs `stress-ng`. The manifest already defines where the test should run in the cluster i.e. in which namespace. The functional unit test is time-bound in this case and scoped to 15 minutes. Therefore, we deploy this test, wait for 15 minutes, then delete the manifest to end the loop. The test instructions depend on the functional unit of each CNCF project.
+
+In the example above, the Kubernetes manifest that is applied to the cluster is located in a different repository, but this could be located anywhere. This is due to preferences for CNCF project maintainers who may have a own repository with configuration related to the Green Reviews pipeline, as is the case with the Falco project.
+
+Each workflow should ensure that any artefacts that were deployed as part of the test instructions should be deleted at the end of the test run.
+
+### Authentication
+
+Before the test workflow is called on, we assume that the workflow already contains a secret with a kubeconfig to authenticate with the test cluster and Falco has already been deployed to it. It is required that the pipeline authenticates with the Kubernetes cluster before running the job with the test.
```

Comment:
Is this taken care of in proposal 1?

Reply:
Yes but every job start a new VM so we need to authenticate again for every job who need to interact with the cluster. I think the test workflow should take as input a kubeconfig and use it to authenticate.

Reply:
Yes, that was the idea we had as well @SFulpius 👍

Reply:
@SFulpius not sure if I get it by `every job start a new VM`, AFAIR We don't create Infra for every job. Actually we are storing the kubeconfig as a secret and utilizing (with special perms ;) )
cc @rossf7

Reply:
@SFulpius not sure about `every job start a new VM`, AFAIR infra part is only touched at main branch changes and (cp: 1, wp: 2). we are getting the Kubeconfig only from the gh.secrets. not sure if we need to update the tokens for the kubeconfig, TTL in k3s is  no defined ;)

cc @rossf7

Reply:
@dipankardas011 
What I wanted to say is that every Github Actions job runs in a new VM, thus we need to authenticate in each job interacting with the cluster.

Reply:
Yes, using gh secrets 👍

Reply:
Yeah sorry my bad, it was already written [here](https://github.com/cncf-tags/green-reviews-tooling/pull/93/files#:~:text=benchmark%2Dpipeline.yaml-,jobs%3A,%60%60%60,-This%20will%20fetch), all good!